projects/step-frontend/src/lib/modules/timeseries/pages/dashboard/chart-dashlet/chart-dashlet.component.ts

```diff
@@ -0,0 +1,293 @@
+          this.context.getFilteringSettings().filterItems,
+          this.item.chartSettings!.filters
+        ),
+      ];
+    }
+    return FilterUtils.filtersToOQL(filterItems, 'attributes');
+  }
+
+  openChartSettings(): void {
+    this._matDialog
+      .open(ChartDashletSettingsComponent, { data: { item: this.item } })
+      .afterClosed()
+      .subscribe((updatedItem) => {
+        if (updatedItem) {
+          Object.assign(this.item, updatedItem);
+          this.refresh().subscribe();
+        }
+      });
+  }
+
+  private createChart(response: TimeSeriesAPIResponse): void {
+    const groupDimensions = this.getChartGrouping();
+    const xLabels = TimeSeriesUtils.createTimeLabels(response.start, response.end, response.interval);
+    const primaryAxes = this.item.chartSettings!.primaryAxes!;
+    const primaryAggregation = this.selectedAggregate!;
+    const series: TSChartSeries[] = response.matrix.map((series, i) => {
+      const labelItems = this.getSeriesKeys(response.matrixKeys[i], groupDimensions);
+      const seriesKey = labelItems.join(' | ');
+      const color =
+        primaryAxes.renderingSettings?.seriesColors?.[seriesKey] || this.context.keywordsContext.getColor(seriesKey);
+
+      return {
+        id: seriesKey,
+        label: seriesKey,
+        labelItems: labelItems,
+        legendName: seriesKey,
+        data: series.map((b) => this.getBucketValue(b, primaryAggregation!)),
+        value: (self, x) => TimeSeriesConfig.AXES_FORMATTING_FUNCTIONS.bigNumber(x),
+        stroke: color,
+        fill: (self: uPlot, seriesIdx: number) => UPlotUtils.gradientFill(self, color),
+        points: { show: false },
+        show: true,
+      };
+    });
+    const primaryUnit = primaryAxes.unit!;
+    const yAxesUnit = this.getUnitLabel(primaryAggregation, primaryUnit);
+
+    this._internalSettings = {
+      title: `${this.item.name} (${primaryAggregation})`,
+      xValues: xLabels,
+      series: series,
+      tooltipOptions: {
+        enabled: true,
+        yAxisUnit: yAxesUnit,
+      },
+      showLegend: groupDimensions.length > 0, // in case it has grouping, display the legend
+      axes: [
+        {
+          size: TimeSeriesConfig.CHART_LEGEND_SIZE,
+          scale: 'y',
+          values: (u, vals) => {
+            return vals.map((v: any) => this.getAxesFormatFunction(primaryAggregation, primaryUnit)(v));
+          },
+        },
+      ],
+    };
+  }
+
+  private fetchDataAndCreateChart(): Observable<TimeSeriesAPIResponse> {
+    const settings = this.item.chartSettings!;
+    const groupDimensions = this.getChartGrouping();
+    const request: FetchBucketsRequest = {
+      start: this.context.getSelectedTimeRange().from,
+      end: this.context.getSelectedTimeRange().to,
+      groupDimensions: groupDimensions,
+      oqlFilter: this.composeRequestFilter(settings.metricKey!),
+      numberOfBuckets: 100,
+      percentiles: this.getChartPclToRequest(settings.primaryAxes!.aggregation!),
```

Comment:
Generally I don't like the abreviation PCL, is it for percentile? Especially the naming of this function I don't understand, why is it called "ToRequest"?